diff -upr lsof_4.88-rh.orig/dialects/linux/dnode.c lsof_4.88-rh/dialects/linux/dnode.c
--- lsof_4.88-rh.orig/dialects/linux/dnode.c	2014-10-14 00:26:03.000000000 +0200
+++ lsof_4.88-rh/dialects/linux/dnode.c	2015-01-05 19:57:05.562294509 +0100
@@ -580,7 +580,7 @@ process_proc_node(p, pbr, s, ss, l, ls)
 	    Lf->inp_ty = 1;
 
 #if	defined(HASEPTOPTS)
-	    if (Lf->ntype == N_FIFO) {
+	    if ((Lf->ntype == N_FIFO) && FpipeE) {
 	    	(void) enter_pinfo();
 		Lf->sf |= SELPINFO;
 	    }
